com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/utils/WorkspaceFileService.java

```diff
@@ -0,0 +1,254 @@
+        .map(it -> Paths.get(outer, it))
+        .filter(Files::exists)
+        .collect(toList());
+  }
+
+  @Override
+  public Path createFolders(Path path) {
+    try {
+      return Files.createDirectories(path);
+    } catch (IOException e) {
+      log.error("Error creating folders for " + path, e);
+      return null;
+    }
+  }
+
+  @Override
+  public void createFolder(@Nonnull Path path) {
+    try {
+      Files.createDirectory(path);
+    } catch (IOException e) {
+      log.error("Error creating folder: " + path, e);
+    }
+  }
+
+  @Override
+  public void createFile(@Nonnull Path path) {
+    if (!path.toFile().exists()) {
+      try {
+        Files.createFile(path);
+      } catch (IOException e) {
+        log.error("Error creating file: " + path, e);
+      }
+    }
+  }
+
+  @Nullable
+  @Override
+  public String getNameFromURI(@Nonnull String uri) {
+    try {
+      return FilenameUtils.getBaseName(Paths.get(new URI(uri)).getFileName().toString());
+    } catch (URISyntaxException e) {
+      log.error("Cannot get file name from: " + uri, e);
+      return null;
+    }
+  }
+
+  @Nullable
+  @Override
+  public String getExtensionFromURI(@Nonnull String uri) {
+    return FilenameUtils.getExtension(uri);
+  }
+
+  @Nonnull
+  @Override
+  public List<String> getContentFromFile(@Nonnull Path filePath) {
+    try {
+      return readAllLines(filePath);
+    } catch (IOException e) {
+      log.error("Cannot get content of: " + filePath.toString(), e);
+      return emptyList();
+    }
+  }
+
+  @Nullable
+  @Override
+  public String getContentByPath(@Nonnull Path path) {
+    try (Stream<String> stream = Files.lines(path)) {
+      return stream.reduce((s1, s2) -> s1 + "\r\n" + s2).orElse("");
+    } catch (IOException e) {
+      log.error("Cannot get content of: " + path.toString(), e);
+      return null;
+    }
+  }
+
+  @Nullable
+  @Override
+  public Path applySearch(@Nonnull String fileName, @Nonnull Path targetFolderPath) {
+    try (Stream<Path> pathStream =
+        Files.find(
+            targetFolderPath,
+            100,
+            (path, basicFileAttributes) -> isValidFileFound(path.toFile(), fileName),
+            FileVisitOption.FOLLOW_LINKS)) {
+      return pathStream.findAny().orElse(null);
+    } catch (IOException e) {
+      log.error(format("Error searching for %s in %s", fileName, targetFolderPath.toString()), e);
+      return null;
+    }
+  }
+
+  @Override
+  @Nullable
+  public Path getPathFromURI(@Nonnull String uri) {
+    try {
+      return Paths.get(new URI(uri).normalize());
+    } catch (URISyntaxException e) {
+      log.error("Cannot find file by given URI: " + uri, e);
+      return null;
+    }
+  }
+
+  @Override
+  public void removeIfPresent(@Nonnull String element, @Nonnull Path targetPath) {
+    try {
+      List<String> result = getContentFromFile(targetPath);
+
+      List<String> updatedLines = result.stream().filter(s -> !s.equals(element)).collect(toList());
```

Comment:
It makes sense to use Set here instead of List.

Reply:
depends on caseinsensitive, or  equalsIgnoreCase  usage